### PR TITLE
Fix #3951 - OpenMetadata Airflow REST

### LINKED ISSUE #3951: Airflow REST APIs refactor

### PR DESCRIPTION
**Pull request by pmbrull**
### Describe your changes :

This PR fixes #3951

This is the first round of the implementation, using the custom package directly for the ingestion Docker.

There's an example that you can send to the endpoint and a DAG should be created. It is failing to run for some arguments etc, but the important part is that the DAG exists and runs, together with the callbacks.

#
### Type of change :

- [x] Bug fix
- [x] Improvement
- [x] New feature
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/open-source-community/developer) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: ingestion/Dockerfile [hunk 1]
```diff
@@ -1,27 +1,38 @@
...
+FROM base as airflow
 ENV AIRFLOW_VERSION=2.1.4
 ENV CONSTRAINT_URL="https://raw.githubusercontent.com/apache/airflow/constraints-${AIRFLOW_VERSION}/constraints-3.9.txt"
 # Add docker provider for the DockerOperator
 RUN pip install "apache-airflow[docker]==${AIRFLOW_VERSION}" --constraint "${CONSTRAINT_URL}"
+
+
+FROM airflow as apis
+WORKDIR /openmetadata-airflow-apis
+COPY openmetadata-airflow-apis /openmetadata-airflow-apis
```
Comment (pmbrull) [inline ingestion/Dockerfile:18]: Build ingestion container with the repo apis code

File: openmetadata-airflow-apis/README.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+## Using the API
+Once you deploy the plugin and restart the webserver, you can start to use the REST API. Bellow you will see the endpoints that are supported. 
+
+**Note:** If enable RBAC, `http://{AIRFLOW_HOST}:{AIRFLOW_PORT}/rest_api/`<br>
+This web page will show the Endpoints supported and provide a form for you to test submitting to them.
+
+- [deploy_dag](#deploy_dag)
+- [delete_dag](#delete_dag)
+
+### ***<span id="deploy_dag">deploy_dag</span>***
```
Comment (harshach) [inline openmetadata-airflow-apis/README.md:110]: @pmbrull trigger dag is very important so is list of dag statuses. We don't want to switch between these apis and airflow apis. Lets add them here

Comment (pmbrull) [reply openmetadata-airflow-apis/README.md:110]: will do, also the check connection

**Comment by pmbrull** [comment]
will merge it in after checking the CI, must have broken something

**Pull request merged**